src/Microsoft.TemplateEngine.Cli/CommandParsing/FilterOptions.cs

```diff
@@ -0,0 +1,92 @@
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli.TemplateResolution;
+using Microsoft.TemplateEngine.Edge.Template;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Cli.CommandParsing
+{
+    internal class FilterOption
+    {
+        internal string Name { get; set; }
+        internal Func<INewCommandInput, string> FilterValue { get; set; }
+        internal Func<INewCommandInput, bool> IsFilterSet { get; set; }
+        internal Func<INewCommandInput, Func<ITemplateInfo, MatchInfo?>> TemplateMatchFilter { get; set; }
+        internal Func<ListOrHelpTemplateListResolutionResult, bool> MismatchCriteria { get; set; }
+    }
+    internal static class SupportedFilterOptions
+    {
+
+        internal static readonly IReadOnlyCollection<FilterOption> SupportedListFilters = new List<FilterOption>()
+        {
+            SupportedFilterOptions.AuthorFilter,
+            SupportedFilterOptions.BaselineFilter,
+            SupportedFilterOptions.LanguageFilter,
+            SupportedFilterOptions.TypeFilter
+        };
+
+        internal static FilterOption BaselineFilter
+        {
+            get
+            {
+                return new FilterOption()
```

Comment:
Consider a private method that consolidates the parametrized logic to build `FilterOption` instances for `BaselineFilter`, `AuthorFilter` & others.  In the future, the duplicate logic could get out of sync.

Reply:
not sure I get the suggestion. Are you suggesting to implement the logic in constructor instead based on command property getter and name?